src/raftstore/coprocessor/region_info_accessor.rs

```diff
@@ -0,0 +1,753 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::sync::{mpsc, Arc, Mutex};
+use std::usize;
+
+use super::{
+    Coprocessor, CoprocessorHost, ObserverContext, RegionChangeEvent, RegionChangeObserver,
+    RoleObserver,
+};
+use kvproto::metapb::Region;
+use raft::StateRole;
+use raftstore::store::keys::data_end_key;
+use util::collections::HashMap;
+use util::worker::{Builder as WorkerBuilder, Runnable, Scheduler, Worker};
+
+const CHANNEL_BUFFER_SIZE: usize = usize::MAX; // Unbounded
+
+/// `RegionInfoAccessor` is used to collect all regions on this TiKV into a collection so that other
+/// parts of TiKV can get region information from it. It registers a observer to raftstore, which
+/// is named `RegionEventListener`, and it simply send some specific types of events through a channel.
+/// In the mean time, `RegionCollection` keeps fetching messages from the channel, and mutate
+/// the collection according tho the messages. When an accessor method of `RegionInfoAccessor` is
+/// called, it also simply send a message to `RegionCollection`, and the result will be send
```

Comment:
s/tho/to
s/RegionCollection/RegionInfoAccessor

Reply:
It indeed send the event to RegionCollection..